wetterdienst/dwd/radar/access.py

```diff
@@ -0,0 +1,329 @@
+        radar_data_type: Some radar data are available in different data types
+        prefer_local: boolean if file should be read from local store instead
+        write_file: boolean if file should be stored on the drive
+        folder: path for storage
+
+    Returns:
+        list of tuples of a datetime and the corresponding file in bytes
+    """
+
+    if parameter == RadarParameter.RADOLAN_GRID:
+
+        if time_resolution not in (
+            TimeResolution.HOURLY,
+            TimeResolution.DAILY,
+            TimeResolution.MINUTE_5,
+            TimeResolution.MINUTE_15,
+        ):
+            raise ValueError("Wrong TimeResolution for RadarData")
+
+        return _collect_radolan_grid_data(
+            date_times=date_times,
+            time_resolution=time_resolution,
+            prefer_local=prefer_local,
+            write_file=write_file,
+            folder=folder,
+        )
+
+    else:
+        return _collect_generic_radar_data(
+            parameter=parameter,
+            date_times=date_times,
+            radar_site=radar_site,
+            radar_data_type=radar_data_type,
+        )
+
+
+def _collect_generic_radar_data(
+    parameter: RadarParameter,
+    date_times: List[datetime],
+    radar_site: Optional[RadarSite] = None,
+    radar_data_type: Optional[RadarDataType] = None,
+) -> List[Tuple[datetime, BytesIO]]:
+    """
+    Collect raw radar data for COMPOSITE and SITES.
+
+    Remark: As of now, only the LATEST file (like "raa00-dx_10132-latest-boo---bin")
+            will be acquired.
+
+    Args:
+        parameter: What type of radar data should be collected
+        date_times: list of datetime objects for which RADOLAN shall be acquired
+        radar_site:      Site of the radar if parameter is one of RADAR_PARAMETERS_SITES
+        radar_data_type: Some radar data are available in different data types
+    Returns:
+        list of tuples of a datetime and the corresponding file in bytes
+    """
+
+    # data = []
+
+    # FIXME: Implement indexing by timestamp when needed.
+    if date_times[0] != RadarDate.LATEST.value:
+        raise NotImplementedError(
+            "Acquisition of generic radar data only supports LATEST file for now"
+        )
+
+    file_index = create_fileindex_radar(
+        parameter=parameter,
+        radar_site=radar_site,
+        radar_data_type=radar_data_type,
+    )
+
+    # Find latest file.
+    filenames = file_index["FILENAME"].tolist()
+
+    try:
+        latest_file = list(filter(lambda x: "-latest-" in x, filenames))[0]
+    except IndexError:
+        latest_file = filenames[-1]
+
+    # Make up single-entry response.
+    # TODO: Discuss which datetime to use here.
+    # TODO: Maybe decode timestamp from file header, e.g. RX272355, RW272250, DX280005
+    now = datetime.now()
+    payload = _download_generic_data(latest_file)
+    response = [(now, payload)]
```

Comment:
By now, `_collect_generic_radar_data` only implements acquisition of `-latest-` files.

Reply:
> need to implement full indexing by timestamp for `/weather/radar/{composit,radolan,sites}`

5e9073d might solve it. At least, there is a test for **dx reflectivity** already.